circuits/src/transaction.rs

```diff
@@ -131,27 +129,30 @@ impl<const H: usize> TxInputNote<H> {
 pub struct TxOutputNote {
     value: u64,
     value_commitment: JubJubAffine,
-    blinding_factor: JubJubScalar,
+    value_blinder: JubJubScalar,
+    sender_enc: [(JubJubAffine, JubJubAffine); 2],
```

Comment:
it feels odd to have `sender_enc` here and not in WitnessTxOutputNote. I don't have a strong preference, but I maybe would keep it as an independent parameter.

Reply:
I decided to remove the `append_to_circuit` method all together and let the different gadgets append the fields as they need them.